packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -46,13 +46,14 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
             if (templateLiteral.expressions.length) {
               const pattern = buildGlobPattern(templateLiteral)
               // Note: native import.meta.url is not supported in the baseline
-              // target so we use window.location here -
+              // target so we use the global location here. It can be
+              // window.location or self.location in case it is used in a Web Worker.
               s.overwrite(
                 index,
                 index + exp.length,
                 `new URL(import.meta.globEagerDefault(${JSON.stringify(
                   pattern
-                )})[${rawUrl}], window.location)`
+                )})[${rawUrl}], location)`
```

Comment:
Shouldn't we use `self` here? 
It is [defined in both contexts](https://developer.mozilla.org/en-US/docs/Web/API/Window/self), and it may be safer in case someone has a variable called `location` in scope. @lukastaegert do you see an issue with this?
```suggestion
                )})[${rawUrl}], self.location)`
```

Reply:
True, I did not consider local scoping. Though the same argument could be true for `self` 😉. But I agree that that would be probably less likely.

Reply:
Is self == window ?

Reply:
Yes, check the mdn docs I linked above